### ISSUE TITLE
Listen to action finish

### ISSUE THREAD

**Issue by Globegitter** [tj/commander.js#352]
It would be great to have something like `program.on('finish', function(){});` To make it easier to run an action you want to have once all actions/commands are finished, regardless of which one it is, such as clean-up actions, or make it easier to test your whole tool, but just returning promise once everything is finished.

This is just a proposal to simplify certain use-cases.

**Comment by SomeKittens** [comment]
Interesting.  Where would this be useful where `process.on('beforeExit'` wouldn't be (or for tests, an `after` call)?

**Comment by Globegitter** [comment]
The general idea was to have all the command definitions in a cli.js and then once they are done resolve a promise, so the cli can be used in a similar manner as to how ember-cli does (they do not use commander.js).

See here https://github.com/ember-cli/ember-cli/blob/master/tests/acceptance/generate-test.js#L67 and here https://github.com/ember-cli/ember-cli/blob/master/tests/helpers/ember.js#L18 for example. Not sure if `process.on('beforeExit')` or `after` really works that well in such use-cases but I will also speond some more time looking into that and other ways to do that.

**Comment by shadowspawn** [comment]
Interesting comments thanks.

This issue has not received many reactions or links, and has not had any activity in over six months. It isn't likely to get acted on due to this report.

Feel free to open a new issue if it comes up again, with new information and renewed interest.
 
Thank you  for your contributions.
